server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.node;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.Version;
+import org.elasticsearch.action.admin.cluster.node.stats.NodeStats;
+import org.elasticsearch.action.admin.indices.stats.CommonStatsFlags;
+import org.elasticsearch.cluster.ClusterChangedEvent;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateListener;
+import org.elasticsearch.cluster.DiskUsage;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.routing.ShardRoutingState;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.health.HealthStatus;
+import org.elasticsearch.health.metadata.HealthMetadata;
+import org.elasticsearch.health.node.selection.HealthNode;
+import org.elasticsearch.health.node.selection.HealthNodeTaskExecutor;
+import org.elasticsearch.monitor.fs.FsInfo;
+import org.elasticsearch.node.NodeService;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+
+/**
+ * This class monitors the health of the node regarding the load on several resources.
+ * Currently, it only checks for available disk space. Furthermore, it informs the health
+ * node about the local health upon change or when a new node is detected.
+ */
+public class LocalHealthMonitor implements ClusterStateListener {
+
+    private static final Logger logger = LogManager.getLogger(LocalHealthMonitor.class);
+
+    public static final Setting<TimeValue> INTERVAL_SETTING = Setting.timeSetting(
```

Comment:
Should we document this new setting?

Reply:
I have a task in https://github.com/elastic/elasticsearch/issues/84811 to document all settings introduced as part of this and to add the troubleshooting page. I prefer to do all documentation at the end to avoid going live before we remove the feature flag.